scripts/tybalt_vae.py

```diff
@@ -0,0 +1,320 @@
+# 
+# Much of this script is inspired by the [keras variational_autoencoder.py example](https://github.com/fchollet/keras/blob/master/examples/variational_autoencoder.py)
+
+# In[1]:
+
+import os
+import numpy as np
+import pandas as pd
+import matplotlib.pyplot as plt
+
+import tensorflow as tf
+from keras.layers import Input, Dense, Lambda, Layer, Activation
+from keras.layers.normalization import BatchNormalization
+from keras.models import Model
+from keras import backend as K
+from keras import metrics, optimizers
+from keras.callbacks import Callback
+import keras
+
+import pydot
+import graphviz
+from keras.utils import plot_model
+from IPython.display import SVG
+from keras.utils.vis_utils import model_to_dot
+
+
+# In[2]:
+
+print(keras.__version__)
+tf.__version__
+
+
+# ## Load Functions and Classes
+# 
+# This will facilitate connections between layers and also custom hyperparameters
+
+# In[3]:
+
+# Function for reparameterization trick to make model differentiable
+def sampling(args):
+    
+    import tensorflow as tf
+    # Function with args required for Keras Lambda function
+    z_mean, z_log_var = args
+
+    # Draw epsilon of the same shape from a standard normal distribution
+    epsilon = K.random_normal(shape=tf.shape(z_mean), mean=0.,
+                              stddev=epsilon_std)
+    
+    # The latent vector is non-deterministic and differentiable
+    # in respect to z_mean and z_log_var
+    z = z_mean + K.exp(z_log_var / 2) * epsilon
+    return z
+
+
+class CustomVariationalLayer(Layer):
+    """
+    Define a custom layer that learns and performs the training
+
+    """
+    def __init__(self, **kwargs):
+        # https://keras.io/layers/writing-your-own-keras-layers/
+        self.is_placeholder = True
+        super(CustomVariationalLayer, self).__init__(**kwargs)
+
+    def vae_loss(self, x_input, x_decoded):
+        reconstruction_loss = original_dim * metrics.binary_crossentropy(x_input, x_decoded)
+        kl_loss = - 0.5 * K.sum(1 + z_log_var_encoded - K.square(z_mean_encoded) - 
+                                K.exp(z_log_var_encoded), axis=-1)
+        return K.mean(reconstruction_loss + (K.get_value(beta) * kl_loss))
+
+    def call(self, inputs):
+        x = inputs[0]
+        x_decoded = inputs[1]
+        loss = self.vae_loss(x, x_decoded)
+        self.add_loss(loss, inputs=inputs)
+        # We won't actually use the output.
+        return x
```

Comment:
Even if I'm not using the output, Keras requires on to be supplied. If I don't include it, and try to `vae.compile()` I get this error:

```
/home/gway/anaconda3/lib/python3.5/site-packages/keras/engine/topology.py in _add_inbound_node(self, input_tensors, output_tensors, input_masks, output_masks, input_shapes, output_shapes, arguments)
    690         # Update tensor history, _keras_shape and _uses_learning_phase.
    691         for i in range(len(output_tensors)):
--> 692             output_tensors[i]._keras_shape = output_shapes[i]
    693             uses_lp = any([getattr(x, '_uses_learning_phase', False) for x in input_tensors])
    694             uses_lp = getattr(self, 'uses_learning_phase', False) or uses_lp

AttributeError: 'NoneType' object has no attribute '_keras_shape'
```

Furthermore, I borrow this function from the [VAE example provided](https://github.com/fchollet/keras/blob/master/examples/variational_autoencoder.py#L45).

I do state `Much of this script is inspired by the keras variational_autoencoder.py example` in the opening docstring, but I will add something similar to the function docstring too to reduce confusion